### DISCUSSION TITLE
Watch for append(), remove() on arrays

### DISCUSSION THREAD

**Discussion by andfk** [react-hook-form/react-hook-form#3530] [Q&A]
Hi, this library is amazing but im stuck on trying to detect new appended/removed items on an array list.

I've tested with a `watch()` like is suggested in multiple places but that triggers onChange events too, which is something i want to skip (i prefer only onBlur). Is this behaviour allowed or should i consider a PR?

**Comment by bluebill1049** [comment]
Is this a perf issue? maybe you can consider `useWatch` to isolate at component level?

**Reply by andfk** [reply]
Hi @bluebill1049, thanks for your quick reply.

Do you mean to create a `useWatch` in each item of an array list? Following that direction not sure if im able to listen for the `.append()`/`.remove()` events/callbacks. I've noticed that a "form" with appended/removed items doesnt trigger "onChange" events. Is it intentional? as the "useFieldArray" gets notified but not the parent form.

**Reply by bluebill1049** [reply]
I mean where you want to subscribe to the change, it would be outside the Field Array. what do you mean trigger `onChange`?

**Reply by andfk** [reply]
I've created a CSB to demonstrate my requirements
CSB: https://codesandbox.io/s/onchange-usefieldarray-vpt57

In resume, i need to be able to notify when items are added/removed in a list and when the values changes too but instead of `onChange`, i prefer `onBlur` due perf reasons. 
- `onBlur` "could work" but is not called on items removed. (check console)
- `onChange` only gets called when field values changes not when items are added/removed.
- `watch` is perfect but every row on a list gets re-rendered due field value changes. 

So my idea could be to propose an `onChange` callback for `useFieldArray` that gets called when the items on the list changes unless another approach exists?

**Reply by bluebill1049** [reply]
hmmm, interesting, some good ideas above, at the same time it will make API for other users. If you have concerns over perf you can always isolated re-render with `useWatch` tho, so only that component contains `useWatch` will trigger the re-render.
